service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/kubernetesNamespace/ControlledAzureKubernetesNamespaceResource.java

```diff
@@ -237,19 +229,43 @@ private Step getGetManagedIdentityStep(
     };
   }
 
+  /**
+   * TODO: when implementing the
+   *
+   * @param flightBeanBag
+   * @param missingIdentityBehavior
+   * @return
+   */
+  @NotNull
+  private DeleteControlledResourceStep getGetManagedIdentityForDeleteStep(
```

Comment:
This is the one resource where requiring steps that are explicitly typed for delete doesn't fit cleanly.

The main options I could see are:
1.  Switch to a different approach (give up).
2. Add the `DeleteControlledResourceStep` interface to `GetWorkspaceManagedIdentityStep`. This could add undesired behavior on other steps.
3. Add a simple extension for the delete step.

Going with 3 adds a bit more complexity here, but less in the long run.

Reply:
Can you explain more what the undesired behavior would be on the other steps?

Reply:
For example, if we were creating a resource, we would probably _need_ to fail if we can't access a particular resource.